File: DictationAssistant/LyricWriter.vb
Imports System.IO
Imports System.Text

Public Class LyricWriter
    Private Lyric As New Dictionary(Of String, Dictionary(Of Long, Object))
    Public ReadOnly IdTag As New Dictionary(Of String, String)
    Public Sub Append(text As String, offest As Long)
        If Not Lyric.ContainsKey(text) Then
            Lyric.Add(text, New Dictionary(Of Long, Object))
        End If
        Lyric.Item(text).Add(offest, Nothing)
    End Sub
    Public Sub WriteTo(out As Stream)
        Using writer As New StreamWriter(out, Encoding.Default)
            For Each pair In IdTag
                writer.WriteLine($"[{pair.Key}:{pair.Value}]")
            Next
            For Each pair In Lyric
                For Each offest In pair.Value.Keys
                    Dim min = offest \ 60000
                    Dim sec = (offest Mod 60000) / 1000
                    writer.Write($"[{min.ToString("D2")}:{sec.ToString("00.00")}]")
                Next
                writer.WriteLine(pair.Key)
            Next
        End Using
    End Sub
End Class

File: DictationAssistant/LyricWriter.cs
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace DictationAssistant
{
    public class LyricWriter
    {
        private Dictionary<string, Dictionary<long, object>> Lyric = new Dictionary<string, Dictionary<long, object>>();
        public readonly Dictionary<string, string> IdTag = new Dictionary<string, string>();
        public void Append(string text, long offest)
        {
            if (!Lyric.ContainsKey(text))
                Lyric.Add(text, new Dictionary<long, object>());
            Lyric[text].Add(offest, null);
        }
        public void WriteTo(Stream output)
        {
            using (StreamWriter writer = new StreamWriter(output, Encoding.Default))
            {
                foreach (var pair in IdTag)
                    writer.WriteLine($"[{pair.Key}:{pair.Value}]");
                foreach (var pair in Lyric)
                {
                    foreach (var offest in pair.Value.Keys)
                    {
                        var min = offest / 60000;
                        var sec = (offest % 60000) / (double)1000;
                        writer.Write($"[{min.ToString("D2")}:{sec.ToString("00.00")}]");
                    }
                    writer.WriteLine(pair.Key);
                }
            }
        }
    }
}
